This patch adds papi support for AMD Family 15h cpus. There are the
following changes compared to family 10h:

* Instead of separate events for data and instruction TLB misses there
  is now an unified event for this (UNIFIED_TLB_MISS).

* Some events and umasks were renamed (CYCLES_FPU_EMPTY,
  RETIRED_SSE_OPS, RETIRED_MMX_FP_INSTRUCTIONS, SSE).

* Floating point instructions has been modified: PAPI_FP_INS,
  PAPI_FP_OPS, PAPI_SP_OPS, PAPI_DP_OPS.

* Multiply-add instructions were included to PAPI_FML_INS,
  PAPI_FAD_INS.

The remaining events did not change.

Note that this requires a kernel that supports perf_event for AMD
family 15h.

Events available:

    Name        Code    Deriv Description (Note)
PAPI_L1_DCM  0x80000000  No   Level 1 data cache misses
PAPI_L1_ICM  0x80000001  No   Level 1 instruction cache misses
PAPI_L2_DCM  0x80000002  No   Level 2 data cache misses
PAPI_L2_ICM  0x80000003  No   Level 2 instruction cache misses
PAPI_L1_TCM  0x80000006  Yes  Level 1 cache misses
PAPI_L2_TCM  0x80000007  No   Level 2 cache misses
PAPI_FPU_IDL 0x80000012  No   Cycles floating point units are idle
PAPI_TLB_DM  0x80000014  No   Data translation lookaside buffer misses
PAPI_TLB_IM  0x80000015  No   Instruction translation lookaside buffer misses
PAPI_TLB_TL  0x80000016  No   Total translation lookaside buffer misses
PAPI_STL_ICY 0x80000025  No   Cycles with no instruction issue
PAPI_HW_INT  0x80000029  No   Hardware interrupts
PAPI_BR_TKN  0x8000002c  No   Conditional branch instructions taken
PAPI_BR_MSP  0x8000002e  No   Conditional branch instructions mispredicted
PAPI_TOT_INS 0x80000032  No   Instructions completed
PAPI_FP_INS  0x80000034  No   Floating point instructions
PAPI_BR_INS  0x80000037  No   Branch instructions
PAPI_VEC_INS 0x80000038  No   Vector/SIMD instructions (could include integer)
PAPI_RES_STL 0x80000039  No   Cycles stalled on any resource
PAPI_TOT_CYC 0x8000003b  No   Total cycles
PAPI_L1_DCH  0x8000003e  Yes  Level 1 data cache hits
PAPI_L2_DCH  0x8000003f  Yes  Level 2 data cache hits
PAPI_L1_DCA  0x80000040  No   Level 1 data cache accesses
PAPI_L2_DCA  0x80000041  No   Level 2 data cache accesses
PAPI_L1_ICH  0x80000049  Yes  Level 1 instruction cache hits
PAPI_L2_ICH  0x8000004a  No   Level 2 instruction cache hits
PAPI_L1_ICA  0x8000004c  No   Level 1 instruction cache accesses
PAPI_L2_ICA  0x8000004d  No   Level 2 instruction cache accesses
PAPI_L1_ICR  0x8000004f  No   Level 1 instruction cache reads
PAPI_L1_TCH  0x80000055  Yes  Level 1 total cache hits
PAPI_L2_TCH  0x80000056  Yes  Level 2 total cache hits
PAPI_L1_TCA  0x80000058  Yes  Level 1 total cache accesses
PAPI_L2_TCA  0x80000059  No   Level 2 total cache accesses
PAPI_FML_INS 0x80000061  No   Floating point multiply instructions (Also includes multiply-add instructions)
PAPI_FAD_INS 0x80000062  No   Floating point add instructions (Also includes subtract and multiply-add instructions)
PAPI_FDV_INS 0x80000063  No   Floating point divide instructions (Counts both divide and square root instructions)
PAPI_FSQ_INS 0x80000064  No   Floating point square root instructions (Counts both divide and square root instructions)
PAPI_FP_OPS  0x80000066  No   Floating point operations
PAPI_SP_OPS  0x80000067  No   Floating point operations; optimized to count scaled single precision vector operations
PAPI_DP_OPS  0x80000068  No   Floating point operations; optimized to count scaled double precision vector operations
-------------------------------------------------------------------------
Of 40 available events, 7 are derived.

Signed-off-by: Robert Richter <robert.richter@amd.com>
---
 src/papi_events.csv |   51 +++++++++++++++++++++++++++++++++++++++++++++++++++
 1 files changed, 51 insertions(+), 0 deletions(-)

diff --git a/src/papi_events.csv b/src/papi_events.csv
index e4334f2..9c6d4d8 100644
--- a/src/papi_events.csv
+++ b/src/papi_events.csv
@@ -162,6 +162,57 @@ PRESET,PAPI_FAD_INS,NOT_DERIVED,RETIRED_SSE_OPERATIONS:SINGLE_ADD_SUB_OPS:DOUBLE
 PRESET,PAPI_FDV_INS,NOT_DERIVED,RETIRED_SSE_OPERATIONS:SINGLE_DIV_OPS:DOUBLE_DIV_OPS:OP_TYPE,NOTE,"Counts both divide and square root instructions"
 PRESET,PAPI_FSQ_INS,NOT_DERIVED,RETIRED_SSE_OPERATIONS:SINGLE_DIV_OPS:DOUBLE_DIV_OPS:OP_TYPE,NOTE,"Counts both divide and square root instructions"
 #
+CPU,AMD64 (Family 15h RevB)
+#
+PRESET,PAPI_L1_ICH,DERIVED_SUB,INSTRUCTION_CACHE_FETCHES,INSTRUCTION_CACHE_REFILLS_FROM_SYSTEM,INSTRUCTION_CACHE_REFILLS_FROM_L2
+PRESET,PAPI_L1_ICM,NOT_DERIVED,INSTRUCTION_CACHE_MISSES
+PRESET,PAPI_L1_ICA,NOT_DERIVED,INSTRUCTION_CACHE_FETCHES
+PRESET,PAPI_L1_ICR,NOT_DERIVED,INSTRUCTION_CACHE_FETCHES
+PRESET,PAPI_L1_DCM,NOT_DERIVED,DATA_CACHE_MISSES:DC_MISS_STREAMING_STORE
+PRESET,PAPI_L1_DCA,NOT_DERIVED,DATA_CACHE_ACCESSES
+PRESET,PAPI_L1_DCH,DERIVED_SUB,DATA_CACHE_ACCESSES,DATA_CACHE_MISSES:DC_MISS_STREAMING_STORE
+PRESET,PAPI_L1_TCA,DERIVED_ADD,DATA_CACHE_ACCESSES,INSTRUCTION_CACHE_FETCHES
+PRESET,PAPI_L1_TCM,DERIVED_ADD,INSTRUCTION_CACHE_MISSES,DATA_CACHE_MISSES:DC_MISS_STREAMING_STORE
+PRESET,PAPI_L1_TCH,DERIVED_POSTFIX,N0|N1|+|N2|-|N3|-|,DATA_CACHE_ACCESSES,INSTRUCTION_CACHE_FETCHES,DATA_CACHE_MISSES:DC_MISS_STREAMING_STORE,INSTRUCTION_CACHE_MISSES
+#
+PRESET,PAPI_L2_ICA,NOT_DERIVED,REQUESTS_TO_L2:INSTRUCTIONS
+PRESET,PAPI_L2_ICM,NOT_DERIVED,L2_CACHE_MISS:INSTRUCTIONS
+PRESET,PAPI_L2_ICH,NOT_DERIVED,INSTRUCTION_CACHE_REFILLS_FROM_L2
+PRESET,PAPI_L2_DCA,NOT_DERIVED,REQUESTS_TO_L2:DATA
+PRESET,PAPI_L2_DCM,NOT_DERIVED,L2_CACHE_MISS:DATA
+PRESET,PAPI_L2_DCH,DERIVED_SUB,REQUESTS_TO_L2:DATA,L2_CACHE_MISS:DATA
+PRESET,PAPI_L2_TCA,NOT_DERIVED,REQUESTS_TO_L2:ALL
+PRESET,PAPI_L2_TCM,NOT_DERIVED,L2_CACHE_MISS:INSTRUCTIONS:DATA
+PRESET,PAPI_L2_TCH,DERIVED_SUB,REQUESTS_TO_L2:INSTRUCTIONS:DATA,L2_CACHE_MISS:ALL
+#
+# not implemented: PRESET,PAPI_L3_TCR,NOT_DERIVED,READ_REQUEST_TO_L3_CACHE:ALL
+# not implemented: PRESET,PAPI_L3_TCM,NOT_DERIVED,L3_CACHE_MISSES:ALL
+# not implemented: PRESET,PAPI_L3_TCH,DERIVED_SUB,READ_REQUEST_TO_L3_CACHE:ALL,L3_CACHE_MISSES:ALL
+#
+PRESET,PAPI_TLB_DM,NOT_DERIVED,UNIFIED_TLB_MISS:4K_DATA:2M_DATA:1GB_DATA
+PRESET,PAPI_TLB_IM,NOT_DERIVED,UNIFIED_TLB_MISS:4K_INST:2M_INST:1G_INST
+PRESET,PAPI_TLB_TL,NOT_DERIVED,UNIFIED_TLB_MISS:ALL
+#
+PRESET,PAPI_BR_INS,NOT_DERIVED,RETIRED_BRANCH_INSTRUCTIONS
+PRESET,PAPI_BR_TKN,NOT_DERIVED,RETIRED_TAKEN_BRANCH_INSTRUCTIONS
+PRESET,PAPI_BR_MSP,NOT_DERIVED,RETIRED_MISPREDICTED_BRANCH_INSTRUCTIONS
+#
+PRESET,PAPI_STL_ICY,NOT_DERIVED,DECODER_EMPTY
+PRESET,PAPI_RES_STL,NOT_DERIVED,DISPATCH_STALLS
+PRESET,PAPI_HW_INT,NOT_DERIVED,INTERRUPTS_TAKEN
+#
+PRESET,PAPI_FPU_IDL,NOT_DERIVED,CYCLES_FPU_EMPTY
+PRESET,PAPI_VEC_INS,NOT_DERIVED,RETIRED_MMX_FP_INSTRUCTIONS:SSE
+PRESET,PAPI_FP_INS,NOT_DERIVED,RETIRED_SSE_OPS:ALL
+PRESET,PAPI_FP_OPS,NOT_DERIVED,RETIRED_SSE_OPS:ALL
+PRESET,PAPI_SP_OPS,NOT_DERIVED,RETIRED_SSE_OPS:SINGLE_ADD_SUB_OPS:SINGLE_MUL_OPS:SINGLE_DIV_OPS:SINGLE_MUL_ADD_OPS
+PRESET,PAPI_DP_OPS,NOT_DERIVED,RETIRED_SSE_OPS:DOUBLE_ADD_SUB_OPS:DOUBLE_MUL_OPS:DOUBLE_DIV_OPS:DOUBLE_MUL_ADD_OPS
+#
+PRESET,PAPI_FML_INS,NOT_DERIVED,RETIRED_SSE_OPS:SINGLE_MUL_OPS:DOUBLE_MUL_OPS:SINGLE_MUL_ADD_OPS:DOUBLE_MUL_ADD_OPS,NOTE,"Also includes multiply-add instructions"
+PRESET,PAPI_FAD_INS,NOT_DERIVED,RETIRED_SSE_OPS:SINGLE_ADD_SUB_OPS:DOUBLE_ADD_SUB_OPS:SINGLE_MUL_ADD_OPS:DOUBLE_MUL_ADD_OPS,NOTE,"Also includes subtract and multiply-add instructions"
+PRESET,PAPI_FDV_INS,NOT_DERIVED,RETIRED_SSE_OPS:SINGLE_DIV_OPS:DOUBLE_DIV_OPS,NOTE,"Counts both divide and square root instructions"
+PRESET,PAPI_FSQ_INS,NOT_DERIVED,RETIRED_SSE_OPS:SINGLE_DIV_OPS:DOUBLE_DIV_OPS,NOTE,"Counts both divide and square root instructions"
+#
 #
 CPU,Intel architectural PMU
 #
-- 
1.7.3.4


_______________________________________________
Perfapi-devel mailing list
Perfapi-devel@eecs.utk.edu
http://lists.eecs.utk.edu/mailman/listinfo/perfapi-devel
